### DISCUSSION TITLE
GDCR 2024

### DISCUSSION THREAD

**Discussion by ReaSu** [swkWien/sessions#136] [General]
# Gedanken zu GDCR 2024, basierend auf GDCR 2023

## Beibehalten von 2023

- Thema für den Tag
- Tetris als Kata in Betracht ziehen?
- Aufstellung der TN von wenig zu viel Erfahrung hat Pairing erleichtert
- Ensemble anbieten
- Hand Up Pattern
- Demo
- Textvorlagen für Aussendungen

## Learnings für 2024

- Diversity braucht bewusstes Investment
- zu Beginn klären:
  - Wer macht mit, und warum?
  - wer möchte/kann wie viel investieren?
  - wer übernimmt welche Rolle?
- Thema/GDCR beim einem Zero Overhead CR durchspielen
- nicht auf 5 Sessions versteifen, sondern vom Thema abhängig machen
- letzte Session als Reflexion des Tages verwenden?
- Zeitplan höher auflösen (Intro, Session, Retro, Pause einzeln anführen, anstatt Brutto-Zeit für Sessions)
- Anmeldung an *einer* Stelle - Eventbrite || Meetup
- WIP Slides gleich public
- Gong um TN herbeizurufen
- Ankündigen, dass wir uns streng an den Zeitplan halten (Pausen nicht überziehen!)
- Social Media Tags auf Slides angeben
- kurzes(!) E-Mail am Vortag mit Link zu Slides, Link zu Kata Bootstraps
- Co-Facilitation nutzen
  - explizit ausmachen, wer in welcher Session mitbetreut
  - evtl. Aufteilung nach Räumen, damit nicht ständig wer daherkommt
  - nutzen, um Thema des Tages anzuleiten
- Feedback von TN einholen
  - was genau wollen wir wissen?

## Überlegungen

- wie organisieren wir die Organisation?
  - Meetings
  - Konversationen: Consider using https://github.com/swkWien/sessions/discussions/126
  - Tools
  - Rollen

**Comment by caichinger** [comment]
Danke für's Anlegen und Festhalten!

**Comment by caichinger** [comment]
Nachträge:
* Ich finde es schön, wenn es in Wien ein physisches Event gibt. Für mich muss das nicht am Tag vom GDCR statt finden. Wenn es für uns einfacher ist, das ein oder zwei Wochen später zu machen, dann machen wir das doch so.
* Notification Settings klären ;)

**Comment by ReaSu** [comment]
> ein oder zwei Wochen später

Guter Punkt. So könnten wir das Herbstferien-Problem umgehen.

**Comment by codecop** [comment]
Ihr seid alle fantastisch

**Comment by paulroho** [comment]
Meine Lieben, hiermit möchte ich anbieten, den GDCR24 als Host zu supporten. Ich würde mich freuen, wieder einmal einen größeren Beitrag in der Community zu leisten. :)

**Comment by caichinger** [comment]
> Meine Lieben, hiermit möchte ich anbieten, den GDCR24 als Host zu supporten. Ich würde mich freuen, wieder einmal einen größeren Beitrag in der Community zu leisten. :)

Cool! :)

**Comment by ReaSu** [comment]
Danke, @paulroho! Ich freu mich schon drauf.

**Comment by caichinger** [comment]
### GDCR24 Team

Hier das Team für alle die mitmachen: https://github.com/orgs/swkWien/teams/gdcr24

**Comment by rolger** [comment]
The date for the GDCR 2024 is officially annouced on https://www.coderetreat.org/?seed=2566. It's as @codecop mention 8. - 9. Nov 2024.

**Comment by caichinger** [comment]
**Claus bei Orga nicht dabei**

Ich nehme mich heuer aus der Organisation raus. Wenn es spontan was geben sollte oder (m)eine Meinung gefragt ist, gerne melden. :)

**Comment by rolger** [comment]
@paulroho und ich habe vereinbart, dass es aus Platz Gründen besser wäre den GDCR 2024 wieder bei Nagarro abzuhalten. Paul's Firma SQUER übersiedelt demnächst und dann gibt es 2025 mehr Platz bei ihnen im Büro. Deswegen heuer noch einmal bei Nagarro und 2025 dann fix bei SQUER im neuen Büro :-)

**Comment by gregorriegler** [comment]
Poah, es ist echt schwer auf GitHub unsere Diskussionen zu finden. Habe diese jetzt sicher 5min lang gesucht, und nur über die Notifikationen gefunden.
Mein learning: In GitHub kann man Diskussionen nur in Repositories anlegen.

GDCR2024 ist sehr bald. 
@rolger du wolltest bei Nagarro hosten.
Steht das?

**Reply by rolger** [reply]
Ja, ich biete es wieder gerne an. 

Leider bin ich beim nächsten kleinen code-retreat auf Urlaub. Wir sollten uns einmal davor abstimmen und die Ideen von vorigem Jahr berücksichtigen

**Comment by JDurstberger** [comment]
Hallo, 

Ich war letztes Jahr beim GDCR und es hat mir sehr gut gefallen.
Ich wollte mich hier nur mal anbieten, falls noch Hilfe bei der Organisation benoetigt wird.

**Comment by gregorriegler** [comment]
![unnamed](https://github.com/user-attachments/assets/537185c9-f217-4ae5-88ae-1a1a5fa4a597)


# Wer
Gregor, Peter, Roland, ?

# Thema für den Tag
Classic mit Fokus auf Iterativ?
- Undecided Customer
- Simple Design

Idee: Mit Guiding Test starten

- Tetris als Kata in Betracht ziehen?
Schwierig?

# Tetris Trivia
https://www.youtube.com/watch?v=dHdYU51pm2o

# Durchprobiert
https://cyber-dojo.org/kata/edit/au4CQq

# Ensemble Vormittag (09:30 - 12:30)
- Teaching
- Start aktivität 
- Ist Demo
- Anfänger abholen

Für Ensemble/Guest Sessions:
Adam?, Jakob?, Kathrin?

Idee: zwei Ensemble Sessions

## 1. Ensemble Session 
- Modus kennen lernen
- Problem kennen lernen
- TDD strikt geführt mit Rotation (e.g. Ping Pong)
- Inkremente nach Wunsch der Teilnehmer
    - Vermutlich zu groß, führt zu Lerneffekt bei 2. Session

## 2. Ensemble Session
- Inkrementell (hint Bild inkrementell vs iterativ)
    - Läuft jederzeit was Sinnvolles
    - Wie man ein größeres Element zerteilt
    - Was ist ein guter erster Testfall
    - Eine Dimension

3 laws of TDD
4 rules of Simple Design

### Facilitation
- Was ist der minimale Business Value?
- Was kann ich noch weglassen?

# Lunch (12:30 - 13:30)

# Classic Pairing Nachmittag (13:30 - 16:30)
- '45 Sessions

Schwierigere Constraints
- Babysteps, TCR
- TDD aiymi


- Aufstellung der TN von wenig zu viel Erfahrung hat Pairing erleichtert

- Hand Up Pattern

- Textvorlagen für Aussendungen
Wir brauchen Text für CIS, Women who code, Eventbrite, SWK, Meetup, etc

**Reply by maosmurf** [reply]
Danke! Ich konnte am Freitag leider nicht dabei sein, aber biete gerne meinen Support für den 9.11. an.
Habt ihr ein follow-up bis dahin geplant?

**Comment by codecop** [comment]
Habe mal die Eckpunkte die wir entschieden haben zusammen gefasst

Motto für uns (Facilitators): Teaching Iterative Development

Sessions
1. Ensemble: TDD kata Intro, by languages
   Gruppen nach gemeinsamen Sprachen bilden

   Ziel
   - Ensemble Modus kennen lernen
   - TDD kennen lernen
   - Problem/Kata kennen lernen

   Facilitation
   - Ensemble strikt geführt mit Rotation
   - Inkremente nach Wunsch der Teilnehmer
     (vermutlich zu groß, führt zu Lerneffekt bei 2. Session)

   Handout für Ensemble Facilitators
   - Ensemble Rules, Rotation auf Zeit
   - Ensemble Facilitation Tipps
   - 3 Laws of TDD
   - Link Wikipedia Tetris

2a. Incremental erklären
   - Bild inkrementell vs. iterativ
   - "Es läuft jederzeit etwas Sinnvolles"
   - Was ist ein guter erster Testfall
   - Wie man ein größeres Element zerteilt
     - zB eine Dimension

2b. Ensemble: Incremental Dev Intro
   neue Gruppen mit anderen Facilitators

   Ziel
   - wollen möglichst kleine Schritte machen

   Handout für Facilitators
   - Was ist der minimale Business Value
   - Was kann ich noch weglassen
   - Ablauf/Beispiel Tests nach Session 12.10.

Lunch

3. Baby Steps (TDD, Pair Programming, Timer)
   Handout für Paare (DONE)

4. TDDaiymi (TDD, Pair Programming)
   Demo (2 Tests und das Duplizierte rausziehen)
   Handout für Paare (DONE)

5. free/TCR without tools (Pair Programming)
   Demo (auch Tests verschwinden bei Revert)
   Handout für Paare (DONE)

Zeitplan attached als Bild
![Zeitplan1](https://github.com/user-attachments/assets/8be81a87-84bf-4e5c-9be4-7f628415de5f)

**Reply by ReaSu** [reply]
Ich hab gerade den Gruppenmodus von cyber-dojo.org entdeckt. Vielleicht könnte das am Nachmittag genutzt werden? Emily hat ein Video dazu, wie man es verwenden kann: https://youtu.be/OGGk-iFVOPQ

**Comment by rolger** [comment]
Ich habe einen 1. Entwurf aud Eventbrite erstellt:

https://www.eventbrite.at/e/global-day-of-coderetreat-vienna-2024-tickets-1047932723987

**Comment by gregorriegler** [comment]
@JDurstberger bitte check dein LinkedIn

**Comment by gregorriegler** [comment]
2 Termine sind per mail raus:
25. Oktober und 4. November 
Wir verwenden unser Gather-Town.

**Comment by codecop** [comment]
Tasks was genau vorzubereiten ist bzw. wer welche Session machen will ist hier #140
